### PR TITLE
Pooja integ test

### PR DESCRIPTION
**Pull request by nathalapooja**
# Description of changes
_How does this change address the problem?_

- Migrated the instance windows Event plugin integration test to GHA

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

Ran the integrationTests on code repo.
```null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Start to generate load in 60.000000 s for the agent to collect and send all the metrics to CloudWatch within the datapoint period
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: System, loglevel: Information, logmsg: System information log
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: System, loglevel: Warning, logmsg: System warning log
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: System, loglevel: Error, logmsg: System error log
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: Application, loglevel: Information, logmsg: Application information log
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: Application, loglevel: Warning, logmsg: Application warning log
null_resource.integration_test (remote-exec): 2023/06/01 20:37:00 Windows Event is successfully created for logname: Application, loglevel: Error, logmsg: Application error log
```

```null_resource.integration_test (remote-exec): 2023/06/01 20:45:02 Start to validate log 'System information log' with number of logs lines 1 within log group i-042323f653def241a, log stream System, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:02 Checking i-042323f653def241a/System
null_resource.integration_test (remote-exec): 2023/06/01 20:45:02 Done paginating log events for i-042323f653def241a/System and found 6 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:02 Start to validate log 'System warning log' with number of logs lines 1 within log group i-042323f653def241a, log stream System, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:02 Checking i-042323f653def241a/System
null_resource.integration_test: Still creating... [10m0s elapsed]
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Done paginating log events for i-042323f653def241a/System and found 6 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Start to validate log 'System error log' with number of logs lines 1 within log group i-042323f653def241a, log stream System, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Checking i-042323f653def241a/System
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Done paginating log events for i-042323f653def241a/System and found 6 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Start to validate log 'Application information log' with number of logs lines 1 within log group i-042323f653def241a, log stream Application, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Checking i-042323f653def241a/Application
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Done paginating log events for i-042323f653def241a/Application and found 3 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Start to validate log 'Application warning log' with number of logs lines 1 within log group i-042323f653def241a, log stream Application, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Checking i-042323f653def241a/Application
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Done paginating log events for i-042323f653def241a/Application and found 3 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Start to validate log 'Application error log' with number of logs lines 1 within log group i-042323f653def241a, log stream Application, start time 2023-06-01 20:42:00 +0000 UTC and end time 2023-06-01 20:43:00 +0000 UTC
null_resource.integration_test (remote-exec): 2023/06/01 20:45:05 Checking i-042323f653def241a/Application
null_resource.integration_test (remote-exec): 2023/06/01 20:45:06 Done paginating log events for i-042323f653def241a/Application and found 3 logs
null_resource.integration_test (remote-exec): 2023/06/01 20:45:06 Test case: win_feature, validate type: feature has been successfully validated
```

### CODE REVIEW

File: validator/validators/basic/basic_validator.go [hunk 1]
```diff
@@ -116,7 +117,9 @@ func (s *BasicValidator) ValidateLogs(logStream, logLine string, numberOfLogLine
 		}
 		actualNumberOfLogLines := 0
 		for _, l := range logs {
-			if strings.Contains(l, logLine) {
+			if logSource == "WindowsEvents" && logLevel != "" && strings.Contains(l, logLine) && strings.Contains(l, logLevel) {
```
Comment (sky333999) [inline validator/validators/basic/basic_validator.go:120]: Given yours is a feature test and even the config being modified is a feature/windows config, should these changes be in the `feature_validator.go` file instead and leave this `basic_validator.go` unmodified?

Comment (SaxyPandaBear) [reply validator/validators/basic/basic_validator.go:120]: I don't know anything about this validator code. Can you explain what the `CheckData` function does and why that is a blocker? I agree that this is a specific feature test and don't think this belongs in a "basic" validator

Comment (nathalapooja) [reply validator/validators/basic/basic_validator.go]: CheckData function validates the metrics and logs being emitted and the implementation is available in basic validator.  Feature validator doesn't have the CheckData function and it uses the basic validator checkData function.

[the remark this answers is not in the document]
Comment (nathalapooja) [reply validator/validators/basic/basic_validator.go]: Updated the code as per the comments.

File: internal/common/logs.go [hunk 2]
```diff
@@ -10,14 +10,44 @@ import (
...
 
 const logLine = "# %d - This is a log line. \n"
 
+func GenerateLogs(configFilePath string, duration time.Duration, sendingInterval time.Duration, logLinesPerMinute int, validationLog []models.LogValidation) error {
+	var multiErr error
+	if err := StartLogWrite(configFilePath, duration, sendingInterval, logLinesPerMinute); err != nil {
+		multiErr = multierr.Append(multiErr, err)
+	}
+	for _, log := range validationLog {
+		if log.LogSource == "WindowsEvents" && log.LogLevel != "" {
```
[the remark this answers is not in the document]
Comment (nathalapooja) [reply internal/common/logs.go]: Updated the code as per the comments.

**Review by sky333999 — APPROVED** [review]

**Pull request merged**